### DISCUSSION TITLE
Request time out error with Google Auth on localhost, works fine in deployment

### DISCUSSION THREAD

**Discussion by sc0rp10n-py** [panva/openid-client#471] [Q&A]
Hi
I am using Next Auth for Google Authentication. When I try to sign up locally I am getting this error
```
[next-auth][error][OAUTH_CALLBACK_ERROR]
https://next-auth.js.org/errors#oauth_callback_error outgoing request timed out after 3500ms {
  error: {
    message: 'outgoing request timed out after 3500ms',
    stack: 'RPError: outgoing request timed out after 3500ms\n' +
      '    at /home/sc0rp10n/fictional-adventure/node_modules/openid-client/lib/helpers/request.js:136:13\n' +
      '    at async Client.grant (/home/sc0rp10n/fictional-adventure/node_modules/openid-client/lib/client.js:1329:22)\n' +
      '    at async Client.callback (/home/sc0rp10n/fictional-adventure/node_modules/openid-client/lib/client.js:474:24)\n' +
      '    at async oAuthCallback (/home/sc0rp10n/fictional-adventure/node_modules/next-auth/core/lib/oauth/callback.js:112:16)\n' +
      '    at async Object.callback (/home/sc0rp10n/fictional-adventure/node_modules/next-auth/core/routes/callback.js:50:11)\n' +
      '    at async NextAuthHandler (/home/sc0rp10n/fictional-adventure/node_modules/next-auth/core/index.js:139:28)\n' +
      '    at async NextAuthNextHandler (/home/sc0rp10n/fictional-adventure/node_modules/next-auth/next/index.js:21:19)\n' +
      '    at async /home/sc0rp10n/fictional-adventure/node_modules/next-auth/next/index.js:57:32\n' +
      '    at async Object.apiResolver (/home/sc0rp10n/fictional-adventure/node_modules/next/dist/server/api-utils.js:101:9)\n' +
      '    at async DevServer.runApi (/home/sc0rp10n/fictional-adventure/node_modules/next/dist/server/next-server.js:320:9)',
    name: 'RPError'
  },
  providerId: 'google',
  message: 'outgoing request timed out after 3500ms'
}
[next-auth][error][CALLBACK_OAUTH_ERROR]
https://next-auth.js.org/errors#callback_oauth_error outgoing request timed out after 3500ms RPError: outgoing request timed out after 3500ms
    at /home/sc0rp10n/fictional-adventure/node_modules/openid-client/lib/helpers/request.js:136:13
    at async Client.grant (/home/sc0rp10n/fictional-adventure/node_modules/openid-client/lib/client.js:1329:22)
    at async Client.callback (/home/sc0rp10n/fictional-adventure/node_modules/openid-client/lib/client.js:474:24)
    at async oAuthCallback (/home/sc0rp10n/fictional-adventure/node_modules/next-auth/core/lib/oauth/callback.js:112:16)
    at async Object.callback (/home/sc0rp10n/fictional-adventure/node_modules/next-auth/core/routes/callback.js:50:11)
    at async NextAuthHandler (/home/sc0rp10n/fictional-adventure/node_modules/next-auth/core/index.js:139:28)
    at async NextAuthNextHandler (/home/sc0rp10n/fictional-adventure/node_modules/next-auth/next/index.js:21:19)
    at async /home/sc0rp10n/fictional-adventure/node_modules/next-auth/next/index.js:57:32
    at async Object.apiResolver (/home/sc0rp10n/fictional-adventure/node_modules/next/dist/server/api-utils.js:101:9)
    at async DevServer.runApi (/home/sc0rp10n/fictional-adventure/node_modules/next/dist/server/next-server.js:320:9) {
  name: 'OAuthCallbackError',
  code: undefined
}
```

There's an issue already created on next auth [here](https://github.com/nextauthjs/next-auth/issues/3920).

Could this be an issue with openid-client itself? If anyone know any fixes then please tell.

**Comment by panva** [comment]
No issue in sight, your Node.js process may not be able to reach google servers for a variety of local network / network setup related reasons.

**Reply by sc0rp10n-py** [reply]
Can you tell more please? I am using WSL for node and this project work.

**Comment by Akhilskt** [comment]
@panva Even i was facing the same issue from past couple of days. I am using this library more than year now, never faced any problem. But suddenly from two days i am seeing this timeout error.I am using open id client both for google and Microsoft. Microsoft auth is working fine. Google is getting failed with this timeout issue from two days. 

error in auth check RPError: outgoing request timed out after 3500ms
    at /home/user/Music/SMA/auth-app/node_modules/openid-client/lib/helpers/request.js:136:13
    at async Client.grant (/home/user/Music/SMA/auth-app/node_modules/openid-client/lib/client.js:1329:22)
    at async Client.callback (/home/user/Music/SMA/auth-app/node_modules/openid-client/lib/client.js:474:24)

At first i though it was some network issue but it looks not
I tried on different devices/OS combinations but same issue. Even i tried with different networks but still issue persists. Even i uninstalled node reinstalled everything but same.
@panva  Could you please shed some more light what to check or how to resolve this

**Reply by panva** [reply]
What you are describing indeed looks like a network issue, not a library issue since, well, nothing changed in the client in the past couple days.

I can hardly shed any light on an issue that's between your localhost and google's endpoints.
